components/calendar/src/chinese_based.rs

```diff
@@ -36,15 +36,12 @@ pub(crate) trait ChineseBasedWithDataLoading: Calendar {
     type CB: ChineseBased;
     /// Get the compiled const data for a ChineseBased calendar; can return `None` if the given year
     /// does not correspond to any compiled data.
-    fn get_compiled_data_for_year(extended_year: i32) -> Option<ChineseBasedCompiledData>;
+    fn get_precomputed_data(&self) -> ChineseBasedPrecomputedData<Self::CB>;
```

Comment:
Suggestion: Add comment that we could add an extra argument such as a pre-existing `YearInfo` here if it would be useful for efficiency.

Reply:
Wrong place, this belongs on PrecomputedDataSource, but yeah, done.